digitalocean/resource_digitalocean_kubernetes_cluster.go

```diff
@@ -345,6 +368,8 @@ func resourceDigitalOceanKubernetesClusterUpdate(ctx context.Context, d *schema.
 			Tags:         expandTags(d.Get("tags").(*schema.Set).List()),
 			AutoUpgrade:  godo.Bool(d.Get("auto_upgrade").(bool)),
 			SurgeUpgrade: d.Get("surge_upgrade").(bool),
+
+			MaintenancePolicy: d.Get("maintenance_policy").(*godo.KubernetesMaintenancePolicy),
```

Comment:
This also needs to call `expandMaintPolicyOpts`.

Also, this line above will need to updated to for the addition:

https://github.com/digitalocean/terraform-provider-digitalocean/blob/24279405c3aca60b167d8531e8332cfa746a0b6b/digitalocean/resource_digitalocean_kubernetes_cluster.go#L364

The SDK has actually grown a new method recently that can make this cleaner:

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema#ResourceData.HasChanges

So it can now look like:

```go
if d.HasChanges("name", "tags", "auto_upgrade", "surge_upgrade", "maintenance_policy") {
```